sensorimotor/sensorimotor/TMRegion.py

```diff
@@ -0,0 +1,571 @@
+                      "during learning.",
+          accessMode='ReadWrite',
+          dataType="UInt32",
+          count=0),
+      permanenceIncrement=dict(
+          description="Amount by which permanences of synapses are "
+                      "incremented during learning.",
+          accessMode='ReadWrite',
+          dataType="Real32",
+          count=1),
+      permanenceDecrement=dict(
+          description="Amount by which permanences of synapses are "
+                      "decremented during learning.",
+          accessMode='ReadWrite',
+          dataType="Real32",
+          count=1),
+      predictedSegmentDecrement=dict(
+          description="Amount by which active permanences of synapses of "
+                      "previously predicted but inactive segments are "
+                      "decremented.",
+          accessMode='ReadWrite',
+          dataType="Real32",
+          count=0),
+      seed=dict(
+          description="Seed for the random number generator.",
+          accessMode='ReadWrite',
+          dataType="UInt32",
+          count=0),
+      tmType=dict(
+          description="Type of tm to use: general",
+          accessMode="ReadWrite",
+          dataType="Byte",
+          count=0,
+          constraints="enum: general"),
+      learnOnOneCell=dict(
+          description="If True, the winner cell for each column will be fixed "
+                      "between resets.",
+          accessMode="ReadWrite",
+          dataType="UInt32",
+          count=1,
+          constraints="bool")))
+
+  # The last group is for parameters that aren't specific to tm
+  otherSpec = dict(
+    learningMode=dict(
+      description="1 if the node is learning (default 1).",
+      accessMode="ReadWrite",
+      dataType="UInt32",
+      count=1,
+      constraints="bool"),
+  )
+
+  return tmSpec, otherSpec
+
+
+
+class TMRegion(PyRegion):
+
+  """
+  """
+
+  def __init__(self,
+               columnDimensions=(2048,),
+               cellsPerColumn=32,
+               activationThreshold=13,
+               initialPermanence=0.21,
+               connectedPermanence=0.50,
+               minThreshold=10,
+               maxNewSynapseCount=20,
+               permanenceIncrement=0.10,
+               permanenceDecrement=0.10,
+               predictedSegmentDecrement = 0.0,
+               seed=42,
+               learnOnOneCell=False,
+               tmImp=getDefaultTMImp(),
+               **kwargs):
+    # Pull out the pooler arguments automatically
+    # These calls whittle down kwargs and create instance variables of TMRegion
+    self._tmClass = getTMClass(tmImp)
+    tmArgTuples = _buildArgs(self._tmClass, self, kwargs)
+
+    # Make a list of automatic pooler arg names for later use
+    self._tmArgNames = [t[0] for t in tmArgTuples]
+
+    # Defaults for all other parameters
+    self.columnDimensions = copy.deepcopy(columnDimensions)
```

Comment:
columnDimensions will be passed as type `numpy.ndarray` so it'll disappear if it's not explicitly copied like this. Another option would be to recast it as a python array instead of copying the ndarray.